Automattic/wp-calypso
# Split SMS/app flows for two-factor authentication setup

#### Changes proposed in this Pull Request

Removes the requirement to enter a phone number when setting up two-factor authentication.

#### Testing instructions



* Log in to a WP.com account without 2fa enabled
* Navigate to /me/security/two-step
* Activate 2fa with an app, or SMS. Make sure it works as usual.

##### Before:

 1. Initial screen <img width="954" alt="Screenshot 2021-10-28 at 20 58 17" src="https://user-images.githubusercontent.com/1081870/139318781-98294766-3d19-4dcd-a0c9-2c9f1374c8c9.png">
 2. Setup screen SMS/App <img width="965" alt="Screenshot 2021-10-28 at 20 58 38" src="https://user-images.githubusercontent.com/1081870/139318789-2236627a-731f-468d-a4f4-f35153b1e506.png">

##### After:
 1. Initial screen <img width="971" alt="Screenshot 2021-10-28 at 20 44 21" src="https://user-images.githubusercontent.com/1081870/139318955-82b4303d-da2e-4786-8360-ddde014c9951.png">
 2. Setup with an app
    * QR code <img width="965" alt="Screenshot 2021-10-28 at 20 44 56" src="https://user-images.githubusercontent.com/1081870/139318957-2c1ed36a-8313-445e-987d-4f22b09c1dc7.png">
    * Backup codes <img width="967" alt="Screenshot 2021-10-28 at 20 51 27" src="https://user-images.githubusercontent.com/1081870/139318959-5ded8c25-72e5-415d-a98f-e940b2b8d449.png">
 3. Setup with SMS
    * Phone number <img width="974" alt="Screenshot 2021-10-28 at 20 51 57" src="https://user-images.githubusercontent.com/1081870/139318963-4360303d-1687-4b81-83cc-58e5340caec1.png">
    * Code verification <img width="964" alt="Screenshot 2021-10-28 at 20 52 18" src="https://user-images.githubusercontent.com/1081870/139318964-3ffc942f-dc4c-4de2-be42-df9f14e261cb.png">
    * Backup codes <img width="973" alt="Screenshot 2021-10-28 at 20 52 39" src="https://user-images.githubusercontent.com/1081870/139318966-d7e12664-a67c-41da-bac5-ac98f50f840e.png">




Related to #36063
